### PR TITLE
Implementing test cases to POST and PATCH user models at the api/users endpoint

### PR DESCRIPTION
**Pull request by cuyakonwu**
This pull request addresses the need for testing of the User API endpoints, specifically focusing on the POST and PATCH functionalities. The introduced tests ensure the reliability and correctness of these critical features, contributing to the overall stability and maintainability of Chigame.

### CODE REVIEW

File: src/chigame/api/tests/test_api.py [hunk 1]
```diff
@@ -282,3 +282,50 @@ def test_delete_message(self):
...
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_user_post(self):
+        user = {"name": "BillyBob", "email": "billy@uchicago.edu"}
+
+        url = reverse("api-user-list")
+        response = self.client.post(url, data=user, format="json")
+
```
Comment (giomhern) [inline src/chigame/api/tests/test_api.py:417]: This looks good to me!

File: src/chigame/api/tests/test_api.py [hunk 2]
```diff
@@ -282,3 +282,50 @@ def test_delete_message(self):
...
+        self.assertEqual(response.data["email"], user["email"])
+        self.assertEqual(response.data["name"], user["name"])
+
+    def test_user_patch(self):
+        user = UserFactory()
+        url = reverse("api-user-detail", kwargs={"pk": user.id})
+
+        updated_data = {"name": "Johnn"}
+
+        response = self.client.patch(url, data=updated_data, format="json")
```
Comment (giomhern) [inline src/chigame/api/tests/test_api.py:429]: Simple, clean, and I approve!

**Review by giomhern — APPROVED** [review]
LGTM!

**Review by majorsylvie — APPROVED** [review]
Nice work! Good to add tests for these, your code uses factories and is straightforward.

Thank you for your work.

**Comment by cuyakonwu** [comment]
This pull request is a step towards fortifying the User API endpoints with comprehensive tests, specifically targeting the POST and PATCH functionalities. The tests validate the reliability and correctness of these critical features, augmenting the overall stability and maintainability of Chigame.

**Key Contributions:**
Created a dedicated test suite for GET at the api/users endpoint, ensuring data retrieval functionality.
Appropriately implemented user list and detail tests, covering a spectrum of scenarios.
Fine-tuned the delete function to ensure the creation of a new user before deletion.
Addressed deliverable tests for both post and patch operations at the api/users endpoint.

**Comment by majorsylvie** [comment]
Issue Score: Excellent

**Pull request merged**